src/setup_payload/AdditionalDataPayloadGenerator.h

```diff
@@ -0,0 +1,47 @@
+/**
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file provides a utility to generate Additional Data payload and its members
+ *      (e.g. rotating device id)
+ *
+ */
+
+#pragma once
+
+#include <core/CHIPError.h>
+#include <system/TLVPacketBufferBackingStore.h>
+
+#include <string>
+
+namespace chip {
+
+class AdditionalDataPayloadGenerator
```

Comment:
Since this class has no data members, should we make it a namespace instead?

Reply:
i thought about that, but since it's a generic class that can be a place for generating other fiends under the additional data, i wanted to create a place holder that can support adding private data members if needed